atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -88,15 +90,45 @@ private fun Project.configureTasks() {
     }
 }
 
-private fun Project.isCompilerPluginAvailable(): Boolean {
-    // kotlinx-atomicfu compiler plugin is available for KGP >= 1.6.20
+private data class KotlinVersion(val major: Int, val minor: Int, val patch: Int)
+
+private fun Project.getKotlinVersion(): KotlinVersion {
     val kotlinVersion = getKotlinPluginVersion()
-    val (majorVersion, minorVersion) = kotlinVersion
+    val (major, minor) = kotlinVersion
         .split('.')
         .take(2)
         .map { it.toInt() }
     val patch = kotlinVersion.substringAfterLast('.').substringBefore('-').toInt()
-    return majorVersion == 1 && (minorVersion == 6 && patch >= 20 || minorVersion > 6)
+    return KotlinVersion(major, minor, patch)
+}
+
+private fun Project.isCompilerPluginAvailable(): Boolean {
+    // kotlinx-atomicfu compiler plugin is available for KGP >= 1.6.20
+    val kotlinVersion = getKotlinVersion()
+    return kotlinVersion.major == 1 && (kotlinVersion.minor == 6 && kotlinVersion.patch >= 20 || kotlinVersion.minor > 6)
+}
+
+private fun Project.applyAtomicfuCompilerPlugin() {
+    val kotlinVersion = getKotlinVersion()
+    // for KGP >= 1.7.20:
+    // compiler plugin for JS IR is applied via the property `kotlinx.atomicfu.enableJsIrTransformation`
+    // compiler plugin for JVM IR is applied via the property `kotlinx.atomicfu.enableJvmIrTransformation`
+    if (kotlinVersion.major == 1 && (kotlinVersion.minor == 7 && kotlinVersion.patch >= 20 || kotlinVersion.minor > 7)) {
```

Comment:
You have at least three duplicating checks, each of them requires non-trivial effort to see if it's correct.
I suggest introducing something like `if (kotlinVersion.atLeast(1, 7, 20)` and use it everywhere